src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -24,44 +25,55 @@ public TemplateEmitter(SemanticModel model)
             this.model = model;
         }
 
+        private string CalculateTemplateHash(string assemblyVersion)
+        {
+            using MemoryStream memoryStream = new();
+            using (var streamWriter = new StreamWriter(memoryStream, UTF8EncodingWithoutBom))
+            using (var writer = new JsonTextWriter(streamWriter))
+            {
+                new TemplateWriter(writer, this.model).Write(true, assemblyVersion, null);
+            }
+            return TemplateHashExtensions.ComputeTemplateHash(UTF8EncodingWithoutBom.GetString(memoryStream.ToArray()));
```

Comment:
Presumably you'll be changing the `ComputeTemplateHash` function to exclude `metadata._generator. templateHash`? Have you thought about what that implementation will look like?

Reply:
`ComputeTemplateHash` effectively calls `TemplateWritter.Write` method with a null templateHash (which omits the field) to calculate the templateHash (first pass: template is written into memory). `TempalteWriter.Write` is then invoked in each of the `Emit` method overloads (second pass: template is written to file system)